### DISCUSSION TITLE
Guidelines for the encoding of spatial data - enumeration

### DISCUSSION THREAD

**Discussion by bfrichet3** [INSPIRE-MIF/helpdesk#96] [Q&A] [answered]
Dear Sir or Madam,

i hope you are all right.

I have a question concerning the way of expressing enumeration values in GML files of one dataset, for instance if I want to express the legalStatus of one object of AdministrativeBoundary type.

Guidelines for the encoding of spatial data Recommendation 19 states that URIs should be used to encode code lsits and their values in INSPIRE code list register and to reference shuch items. If one wants to identify the appropriate codelist value, he should use xlink:href attribute in the concerned xml tag and use the URI of the concept.

There are many enumerations in INSPIRE datasets next to real code lists items. Should the INSPIRE community consider that this Recommendation 19 is relevant for the enumerations values too. As you may know, in Belgium we do have three national languages and I do prefer to identify the enumeration values with their URI's rather than with their human readable value.

Regards
![image](https://user-images.githubusercontent.com/80756452/163546342-cd016723-198a-4bb8-912f-f199837e542e.png)

```

**Comment by sMorrone** [comment]
Dear @bfrichet3,
I fully agree that enumerations are essentially equivalent to non-extensible code lists and should be treated likewise.
In this regard, a proposed amendment to Commission Regulation (EU) No 1089/2010 ([see documents for the 16th meeting of the INSPIRE Committee](https://ec.europa.eu/transparency/comitology-register/screen/documents/076191/1/consult?lang=en)) provides for the removal of the notion of enumerations from the IR text, i.e. in the definitions in Article 2 and elsewhere in the Regulation. Moreover, it is proposed that the explicit mentioning of code list and enumeration values in the IR text is replaced with a reference to the INSPIRE registry, where these values will be managed, under the governance of the MIG.
Clearly, the conversion from enumerations to code lists will affect the currently used xml schemas (which handle code lists and enumerations differently). Implementation details will be discussed when amendment to IR is officially approved (this is expected in the next few months). 
We will share any news as soon as we have.

**Comment by bfrichet3** [comment]
Dear @sMorrone 

Thank you very much for your answer. I will express XML tags correspoding to enumerations attributes as the TG Recommandation 19 states. Maybe it could be usesfull to adapt that Recommandation or to create a new one concerning the enumarations before the IR is ammended.

Regards`

**Comment by sMorrone** [comment, marked as answer]
Unfortunately, if you already provide the enumerations values using the _xlink:href attribute in the concerned xml tag_, your GML dataset will not validate successfully, because of the way the enumerations are currently implemented in the INSPIRE schemas. 
Once the IR is amended, the INSPIRE xml schemas should be amended accordingly to allow for provision of the enumeration values through xlink:href encoding.

**Comment by bfrichet3** [comment]
I think our GML are not yet compliant and this is veryunlikely that they become compliant before the IR's modification. Therefore I will implement these corrections in our next corrections.

Rgards

**Comment by sMorrone** [comment]
Since the purpose of the the technical guidelines repository is the management of the change proposals to the TGs, I am moving this issue to the [general helpdesk ](https://github.com/INSPIRE-MIF/helpdesk)repository. More specifically, I will move it to the "Discussions" tab and classify it as "[Q&A](https://github.com/INSPIRE-MIF/helpdesk/discussions/categories/q-a)"
